### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by dvan2**
Hello Everyone.  I've finished hw3.  I struggle with grouping characters.  I just grouped them based on which family they belonged to.  I also grouped characters that don't have any family.

I've been working a while and barely managed to finished it so I didn't really have the time to check on formatting or deleting unused code or deleting unused resources created by `create-react`.  Let me know if I missed any style guide lines.

Also, I think I did this way more complicated than it should be.  I'm curious to what other people came up with and what people think of mine.

Thanks.

### CODE REVIEW

File: hw3/01-characters.js [hunk 1]
```diff
@@ -1,4 +1,36 @@
...
+
+function createCharacter(img, fullName, title) {
+  //container for character elements
+  const container = document.createElement('div');
+  container.className = 'characterContainer';
+
+  //image
+  const imgElem = document.createElement('img');
+  imgElem.src = img;
+  imgElem.alt = `An image of ${fullName}`;
```
Comment (sieday) [inline hw3/01-characters.js:21]: Good job including this, I totally forgot.

File: hw3/02-charts.js [hunk 2]
```diff
@@ -28,18 +28,109 @@ const borderColors = [
...
+  } else {
+    house.push({
+      name: character.family,
+      count: 1,
+      characterArr: [character],
+    });
+  }
+}
+
+function fixAbnormalData() {
```
Comment (sieday) [inline hw3/02-charts.js:66]: This is a smart way to go about it. I started writing something similar and then figured there must be a simpler way - but I guess with this small of a dataset this works perfectly fine. When it comes to larger datasets I am not sure what the best way is to go about data cleanup.

File: hw3/02-charts.js [hunk 3]
```diff
@@ -28,18 +28,109 @@ const borderColors = [
...
   new Chart(donutChart, {
     type: 'doughnut',
     data: {
-      labels: ['label', 'label', 'label', 'label'],
+      labels: house.map((houseElem) => houseElem.name),
       datasets: [
         {
           label: 'My First Dataset',
-          data: [1, 12, 33, 5],
+          data: house.map((houseElem) => houseElem.count),
```
Comment (sieday) [inline hw3/02-charts.js:133]: Using map here makes a lot of sense, good stuff.

File: hw3/03-myapp/src/pages/Search.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { Character } from './Character';
+
+export function Search() {
+  const [userInput, setUserInput] = React.useState('');
+  const [charactersArr, setCharactersArr] = React.useState([]);
+  const [character, setCharacter] = React.useState([
+    {
+      firstName: 'Welcome',
+      lastName: '',
+      imageUrl:
```
Comment (sieday) [inline hw3/03-myapp/src/pages/Search.js:14]: Love this cute detail :)

**Review by sieday — APPROVED** [review]
This all looks awesome!

I added a few comments, but I do not have any major changes to request. I agree that it feels like there should be a simplified way to go about things, but there is nothing I can personally see to easily remove or swap. 

Thanks!
Sienna

**Review by deanajackson — COMMENTED** [review]
Great work, David! You really applied the skills we've been learning this term and it culminated beautifully.

**Pull request closed without merging**